js/src/hooks/useStoreAddressSynced.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Internal dependencies
+ */
+import useStoreAddress from '.~/hooks/useStoreAddress';
+
+/**
+ * Checks if the store address is synchronized with the Merchant Center (GMC) account address.
+ *
+ * @return {boolean} Returns `true` if the store address matches the GMC account address, otherwise, returns `false`.
+ */
+export default function useStoreAddressSynced() {
+	const { data, loaded } = useStoreAddress();
+
+	if ( ! loaded ) {
+		return false;
+	}
+
+	return data.address && ! Boolean( data.isMCAddressDifferent );
```

Comment:
Instead of checking `data.address` here, which is only part of the address, it's probably better to check the `data.isAddressFilled` property, which checks for all required address fields. See: https://github.com/woocommerce/google-listings-and-ads/blob/99408e1deaf2954f9325f31f2b2ee82b0147119b/js/src/hooks/useStoreAddress.js#L67